### ISSUE TITLE
Remote-SSH Git Pull Push not working No Passphrase in key

### ISSUE THREAD

**Issue by NxP4Code** [microsoft/vscode#94417]
Version: 1.43.2 (user setup)
Commit: 0ba0ca52957102ca3527cf479571617f0de6ed50
Date: 2020-03-24T07:38:38.248Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

Steps to Reproduce:

1. Install SSH-Extension
2. Connect to SSH Host - 
3. Open existing git Repo from remote server which is cloned from our internal git server with ssh URL using RSA Key pair **without passphrase**
4. run git sync from git menu or git pull from terminal 
5. it doesn't finish

```
EXTHOST.LOG
[2020-04-03 21:20:36.021] [exthost] [info] extension host started
[2020-04-03 21:20:36.293] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch {"startup":true,"extensionId":{"value":"vscode.debug-auto-launch","_lower":"vscode.debug-auto-launch"},"activationEvent":"*"}
[2020-04-03 21:20:36.294] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/debug-auto-launch/dist/extension
[2020-04-03 21:20:36.301] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet {"startup":true,"extensionId":{"value":"vscode.emmet","_lower":"vscode.emmet"},"activationEvent":"*"}
[2020-04-03 21:20:36.301] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/emmet/dist/extension
[2020-04-03 21:20:36.334] [exthost] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"extensionId":{"value":"vscode.git","_lower":"vscode.git"},"activationEvent":"*"}
[2020-04-03 21:20:36.334] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/git/dist/main
[2020-04-03 21:20:36.397] [exthost] [info] ExtensionService#_doActivateExtension vscode.github-authentication {"startup":true,"extensionId":{"value":"vscode.github-authentication","_lower":"vscode.github-authentication"},"activationEvent":"*"}
[2020-04-03 21:20:36.397] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/github-authentication/dist/extension.js
[2020-04-03 21:20:36.402] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":true,"extensionId":{"value":"vscode.merge-conflict","_lower":"vscode.merge-conflict"},"activationEvent":"*"}
[2020-04-03 21:20:36.402] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/merge-conflict/dist/extension
[2020-04-03 21:20:36.408] [exthost] [info] ExtensionService#_doActivateExtension vscode.search-result {"startup":true,"extensionId":{"value":"vscode.search-result","_lower":"vscode.search-result"},"activationEvent":"*"}
[2020-04-03 21:20:36.408] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/search-result/dist/extension.js
[2020-04-03 21:20:36.410] [exthost] [info] ExtensionService#_doActivateExtension vscode.vscode-account {"startup":true,"extensionId":{"value":"vscode.vscode-account","_lower":"vscode.vscode-account"},"activationEvent":"*"}
[2020-04-03 21:20:36.410] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/vscode-account/dist/extension.js
[2020-04-03 21:20:36.464] [exthost] [info] ExtensionService#_doActivateExtension vscode.npm {"startup":true,"extensionId":{"value":"vscode.npm","_lower":"vscode.npm"},"activationEvent":"workspaceContains:package.json"}
[2020-04-03 21:20:36.464] [exthost] [info] ExtensionService#loadCommonJSModule file:///c:/Users/<REMOVED>/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/extensions/npm/dist/main
[2020-04-03 21:20:36.601] [exthost] [info] eager extensions activated

```

```
GIT.LOG
Looking for git in: C:\Program Files\Git\cmd\git.exe
Using git 2.26.0.windows.1 from C:\Program Files\Git\cmd\git.exe
> git rev-parse --show-toplevel
> git rev-parse --git-dir
Open repository: e:\REPO
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name master@{u}
> git rev-list --left-right master...refs/remotes/origin/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git config --get commit.template
> git check-ignore -v -z --stdin
> git fetch

```
Both Git-hub Authentication.log and Account.log are empty files.

We also performed debug with our local git server Splunk log debug, When I am running git pull or git synch from VSCODE using the above steps I don't see any requests reaching to git servers.

Same network setup.
1. from local computer PowerShell -> ssh user@remotehost
2. cd repo
3. git pull --> this works fine no issues

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by NxP4Code** [comment]
I have also tried to get help from StackOverflow community.

https://stackoverflow.com/questions/61022777/vscode-remote-ssh-git-pull-sync-not-working-it-hangs-without-errors-ssh

**Comment by roblourens** [comment]
What happens exactly, you say `it doesn't finish`, do you see an error message or something?

**Comment by roblourens** [comment]
Is the key on the local or the remote? Are you expecting ssh agent forwarding to be involved here?

**Comment by NxP4Code** [comment]
I don’t see any errors in the logs , I posted all the logs I see. Let me know if you’re looking for specific logs with steps to generate. I can get it.

**Comment by NxP4Code** [comment]
I have keys are on both local as well as remote I don’t expect agent to be involved since there is no passphrase on the key.

**Comment by NxP4Code** [comment]
When I say it doesn’t finish, I mean on UI it keeps showing it’s trying git action.  But nothing happens further in logs.

**Comment by NxP4Code** [comment]
> Is the key on the local or the remote? Are you expecting ssh agent forwarding to be involved here?

**Comment by NxP4Code** [comment]
@roblourens Please let me know if you need more information.

**Comment by roblourens** [comment]
> or git pull from terminal

This is interesting but since it doesn't work in the terminal, it sounds like maybe an environment issue, not a vscode issue. You could try adding `-vvv` to your git command in the terminal to see if it produces more output.

Also you could try to figure out what the difference in environments is between connecting manually with SSH and in the vscode remote terminal. Same PATH, using the same Git install, is there some environment variable that is supposed to tell it how to find the key? I assume the key is in `~/.ssh` in a default location.

**Comment by NxP4Code** [comment]
Out of desperation, I uninstalled VS Code & Git from both local and host. I installed the VS Code insider, thinking maybe there is something new that might help me fix the issue. But even with an insider release, I have the same behavior. 

I believe its something related to git. I removed git from both host and my local machine. Downloaded new git portable and setup path to mingw64\bin\git.exe on both the machine.


On My host, when I run git command it is recognized since I added mingw64\bin\git.exe in the path. I can do git pull, push like before. 

Local to host -> SSH connection via Powershell --> I can do git pull push. 
![image](https://user-images.githubusercontent.com/56322272/79128420-196f8b00-7d69-11ea-9347-8731d3013f4c.png)


VS Code
![image](https://user-images.githubusercontent.com/56322272/79127767-14f6a280-7d68-11ea-8ba1-883e612a8a85.png)

![image](https://user-images.githubusercontent.com/56322272/79128491-3906b380-7d69-11ea-9314-8bbcf410eccc.png)

Any hint which configuration might be the issue here?

**Comment by NxP4Code** [comment]
I got the git command to work from terminal, path environment variable was not in effect. I restarted host and it worked.

I also tried without keys to make sure it is using the same keys and I am getting error. You can see in the screenshot, In VS COde terminal before adding key I was getting an authentication error. But after adding the correct key it started hanging and doesn't give any errors. Same behavior as the original post.


![image](https://user-images.githubusercontent.com/56322272/79137130-c51fd780-7d77-11ea-94c1-11b0a30aaea4.png)

**Comment by NxP4Code** [comment]
@roblourens ,

I have tried where git and where ssh from ssh shell and terminal from VS code. Both are using the same git and ssh. I have keys set up in .ssh folder in username/.ssh folder that's the only place key is present.

It seems the environment setup is similar in both cases. 

![image](https://user-images.githubusercontent.com/56322272/79152333-a0385e00-7d91-11ea-8597-83ce48209076.png)

![image](https://user-images.githubusercontent.com/56322272/79152629-26ed3b00-7d92-11ea-8667-a4f5e8cb750c.png)


Please let me know if you want me to look into anything specific. I am trying to find a fix for this.

**Comment by roblourens** [comment]
You can run `set` to get a list of environment variables on both sides, any difference?

**Comment by NxP4Code** [comment]
I see the following extra environment variable added when I run Set Command from VS Code terminal, there are other few minor changes but those are expected like added vs code server path in path variable.

I am not sure why VS Code has extra environment variables set, and how does it affect the current issue.


![image](https://user-images.githubusercontent.com/56322272/79228358-e55a9f80-7e26-11ea-98e5-c4f2576498ef.png)

**Comment by roblourens** [comment]
Some are from SSH, some are set by the VS Code terminal, but nothing really stands out to me. I guess I am surprised that some of the basic ones are missing from Windows. Sorry, I'm not really sure what's going on.

**Comment by NxP4Code** [comment]
I dont think any major windows variables are missing, I just attached difference. Sorry my bad, I should have been clear about it.

This is remaining screenshot of the terminal variables. Sorry, couldn't include in one screenshot.

![image](https://user-images.githubusercontent.com/56322272/79253849-7c385380-7e49-11ea-8fe4-20d3cf5bbf50.png)

> How did you launch the other prompt, and did you connect with Remote Desktop, or SSH from a separate terminal or what?

I started command prompt on my local and then ssh into the remote.
